src/main/java/game/Scene.java

```diff
@@ -22,61 +51,104 @@
-            command.getGame().getPlayer()
-                    .listCommands(command.getGame()).forEach(s -> message.append("\t").append(s).append("\n"));
-            message.append("\n\n");
+            } else {
+                callback.onMessage(inspect(command));
+            }
 
-            callback.onMessage(message.toString());
             return true;
-        } else if (command.getAction().equals("search")) {
-            StringBuilder message = new StringBuilder("search:\n");
-            message.append("You find the following items in the scene:\n");
-            items.forEach(i -> message.append("\t").append(i).append("\n"));
-            message.append("\n\n");
+        } else if (command.getAction().equals("search")) { // handles search command
 
-            callback.onMessage(message.toString());
+            callback.onMessage(search(command));
             return true;
-        } else if (actions != null && !command.hasReceiver() && actions.hasCommand(command.getAction())) {
+        } else if (actions != null && !command.hasReceiver() && actions.hasCommand(command.getAction())) { // handles scene action
             // TODO: 04-03-2020 replace with Action.onCommand()
-            Effect effect = actions.getEffect(command.getAction());
-            if (effect != null) {
+            Effects effects = actions.getEffects(command.getAction());
+            if (effects != null) {
 
-                effect.apply(command.getGame());
+                effects.apply(command.getGame(), callback);
 
-                callback.onMessage("Effect applied... " + effect.toString());
+                callback.onMessage("Effect applied... " + effects.toString());
             }
             return true;
-        } else if (command.hasReceiver()) {
-            if (items.contains(command.getReceiver())) {
-                Item item = command.getGame().getItemMap().get(command.getReceiver());
-                item.onCommand(command, callback);
+        } else if (command.hasReceiver() && command.getAction().equals("go")) { // handles next scene action
+            if (nextScenes.containsKey(command.getReceiver())) {
+                command.getGame().setCurrentSceneById(nextScenes.get(command.getReceiver()), callback);
             } else {
-                callback.onMessage("This item does not exist.");
+                callback.onMessage("This direction is not known...");
             }
 
             return true;
+        } else if (command.hasReceiver() && items.contains(command.getReceiver())) { // handles item action
+            return command.getGame().getItemMap().get(command.getReceiver()).handleCommand(command, callback);
+        } else if (command.hasReceiver() && characters.contains(command.getReceiver())) { // handles character action
+            return command.getGame().getCharacterMap().get(command.getReceiver()).handleCommand(command, callback);
         }
 
         return false;
     }
```

Comment:
This is a very long if/else if statement, which isn't great practice because it requires you to ensure that all possible case scenarios and combinations of conditions are covered, especially when the else if conditions contain convoluted boolean expressions.